File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;

        Map<Integer, Integer> map = new HashMap<>();

        // 갯수 Map에 Insert
        for (int i : tangerine) {
            map.put(i, map.getOrDefault(i, 0) + 1);
        }

        List<Integer> keyList = new ArrayList<>(map.keySet());

        // keyList value기준으로 내림차순 정렬
        keyList.sort((o1, o2) -> map.get(o2) - map.get(o1));

        int i = 0;

        while (k > 0) {
            k -= map.get(keyList.get(i));
            answer++;
            i++;
        }

        return answer;
    }
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    list = {}
    for i in tangerine : 
        if i in list : 
            list[i] += 1
        else :
            list[i] = 1
    list = dict(sorted(list.items(), key=lambda x: x[1], reverse = True))
    for i in list : 
        if k<= 0 :
            return answer
        k -= list[i]
        answer += 1
    
    return answer